app/services/charges/validators/volume_service.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+module Charges
+  module Validators
+    class VolumeService < Charges::Validators::BaseService
+      def validate
+        errors = []
+
+        if ranges.blank?
+          errors << :missing_ranges
+          return result.fail!(code: :invalid_properties, message: errors)
+        end
+
+        next_from_value = 0
+        ranges.each_with_index do |range, index|
+          errors << :invalid_amount unless valid_amounts?(range)
+          errors << :invalid_ranges unless valid_bounds?(range, index, next_from_value)
+
+          next_from_value = (range[:to_value] || 0) + 1
```

Comment:
So it supposes that volume charge model only works with integers, right?

I'm not even sure that having decimals volume (from 0 to 1.5 for instance) makes any sense... 😅

Reply:
Well, we could probably handle decimal bounds but we decided not to for now as it make it more complex for bounds validation